File: dotnet/se3-legacy/BigSurTest.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using OpenQA.Selenium.Remote;

namespace Se3Legacy
{
    [TestClass]
    public class BigSurTest : TestBase
    {
        [TestMethod]
        public void LatestChrome()
        {
            DesiredCapabilities caps = new DesiredCapabilities();
            caps.SetCapability("browserName", "chrome");
            caps.SetCapability("platform", "macOS 11.00");
            caps.SetCapability("version", "latest");

            StartDriver(caps);
            ValidateGoogle();
        }
        
        [TestMethod]
        public void LateChrome()
        {
            DesiredCapabilities caps = new DesiredCapabilities();
            caps.SetCapability("browserName", "chrome");
            caps.SetCapability("platform", "macOS 11.00");
            caps.SetCapability("version", "92");

            StartDriver(caps);
            ValidateGoogle();
        }

        [TestMethod]
        public void EarlyChrome()
        {
            DesiredCapabilities caps = new DesiredCapabilities();
            caps.SetCapability("browserName", "chrome");
            caps.SetCapability("platform", "macOS 11.00");
            caps.SetCapability("version", "69");

            StartDriver(caps);
            ValidateGoogle();
        }

        [TestMethod]
        public void LatestEdge()
        {
            DesiredCapabilities caps = new DesiredCapabilities();
            caps.SetCapability("browserName", "MicrosoftEdge");
            caps.SetCapability("platform", "macOS 11.00");
            caps.SetCapability("version", "latest");

            StartDriver(caps);
            ValidateGoogle();
        }

        [TestMethod]
        public void LateEdge()
        {
            DesiredCapabilities caps = new DesiredCapabilities();
            caps.SetCapability("browserName", "MicrosoftEdge");
            caps.SetCapability("platform", "macOS 11.00");
            caps.SetCapability("version", "92");

            StartDriver(caps);
            ValidateGoogle();
        }

        [TestMethod]
        public void EarlyEdge()
        {
            DesiredCapabilities caps = new DesiredCapabilities();
            caps.SetCapability("browserName", "MicrosoftEdge");
            caps.SetCapability("platform", "macOS 11.00");
            caps.SetCapability("version", "79");

            StartDriver(caps);
            ValidateGoogle();
        }

        [TestMethod]
        public void LatestFirefox()
        {
            DesiredCapabilities caps = new DesiredCapabilities();
            caps.SetCapability("browserName", "firefox");
            caps.SetCapability("platform", "macOS 11.00");
            caps.SetCapability("version", "latest");

            StartDriver(caps);
            ValidateGoogle();
        }

        [TestMethod]
        public void LateFirefox()
        {
            DesiredCapabilities caps = new DesiredCapabilities();
            caps.SetCapability("browserName", "firefox");
            caps.SetCapability("platform", "macOS 11.00");
            caps.SetCapability("version", "91");

            StartDriver(caps);
            ValidateGoogle();
        }

        [TestMethod]
        public void EarlyFirefox()
        {
            DesiredCapabilities caps = new DesiredCapabilities();
            caps.SetCapability("browserName", "firefox");
            caps.SetCapability("platform", "macOS 11.00");
            caps.SetCapability("version", "60");

            StartDriver(caps);
            ValidateGoogle();
        }

        [TestMethod]
        public void Safari()
        {
            DesiredCapabilities caps = new DesiredCapabilities();
            caps.SetCapability("browserName", "safari");
            caps.SetCapability("platform", "macOS 11.00");
            caps.SetCapability("version", "14");

            StartDriver(caps);
            ValidateGoogle();
        }
    }
}

File: java/se2-legacy/src/test/java/com/saucelabs/platformconfigurator/se2legacy/BigSurTest.java
package com.saucelabs.platformconfigurator.se2legacy;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.openqa.selenium.remote.DesiredCapabilities;

public class BigSurTest extends AbstractBaseTest {
    @DisplayName("Latest Chrome")
    @Test
    public void chromeLatest() {
        DesiredCapabilities caps = DesiredCapabilities.chrome();
        caps.setCapability("platform", "macOS 11.00");
        caps.setCapability("version", "latest");

        startDriver(caps);
        validateGoogle();
    }

    @DisplayName("Late Chrome")
    @Test
    public void lateChrome() {
        DesiredCapabilities caps = DesiredCapabilities.chrome();
        caps.setCapability("platform", "macOS 11.00");
        caps.setCapability("version", "93");

        startDriver(caps);
        validateGoogle();
    }

    @DisplayName("Early Chrome")
    @Test
    public void earlyChrome() {
        DesiredCapabilities caps = DesiredCapabilities.chrome();
        caps.setCapability("platform", "macOS 11.00");
        caps.setCapability("version", "69");

        startDriver(caps);
        validateGoogle();
    }

    @DisplayName("Latest Edge")
    @Test
    public void latestEdge() {
        DesiredCapabilities caps = DesiredCapabilities.edge();
        caps.setCapability("platform", "macOS 11.00");
        caps.setCapability("version", "latest");

        startDriver(caps);
        validateGoogle();
    }

    @DisplayName("Late Edge")
    @Test
    public void lateEdge() {
        DesiredCapabilities caps = DesiredCapabilities.edge();
        caps.setCapability("platform", "macOS 11.00");
        caps.setCapability("version", "92");

        startDriver(caps);
        validateGoogle();
    }

    @DisplayName("Early Edge")
    @Test
    public void earlyEdge() {
        DesiredCapabilities caps = DesiredCapabilities.edge();
        caps.setCapability("platform", "macOS 11.00");
        caps.setCapability("version", "79");

        startDriver(caps);
        validateGoogle();
    }

    @DisplayName("latest Firefox")
    @Test
    public void latestFirefox() {
        DesiredCapabilities caps = DesiredCapabilities.firefox();
        caps.setCapability("platform", "macOS 11.00");
        caps.setCapability("version", "latest");

        startDriver(caps);
        validateGoogle();
    }

    @DisplayName("late Firefox")
    @Test
    public void lateFirefox() {
        DesiredCapabilities caps = DesiredCapabilities.firefox();
        caps.setCapability("platform", "macOS 11.00");
        caps.setCapability("version", "91");

        startDriver(caps);
        validateGoogle();
    }

    @DisplayName("early Firefox")
    @Test
    public void earlyFirefox() {
        DesiredCapabilities caps = DesiredCapabilities.firefox();
        caps.setCapability("platform", "macOS 11.00");
        caps.setCapability("version", "60");

        startDriver(caps);
        validateGoogle();
    }

    @DisplayName("Safari")
    @Test
    public void safari() {
        DesiredCapabilities caps = DesiredCapabilities.safari();
        caps.setCapability("platform", "macOS 11.00");
        caps.setCapability("version", "14");

        startDriver(caps);
        validateGoogle();
    }
}
